src/PhpSpreadsheet/Writer/Xls/Worksheet.php

```diff
@@ -441,7 +441,23 @@ public function close()
                     case DataType::TYPE_FORMULA:
                         $calculatedValue = $this->preCalculateFormulas ?
                             $cell->getCalculatedValue() : null;
-                        $this->writeFormula($row, $column, $cVal, $xfIndex, $calculatedValue);
+                        if (-3 == $this->writeFormula($row, $column, $cVal, $xfIndex, $calculatedValue)) {
+                            $calculatedValue = $cell->getCalculatedValue();
```

Comment:
Is it necessary to execute `$cell->getCalculatedValue()` a second time after it's already been called in line 442 (albeit conditionally)?

Reply:
Lines 445-447 address this,